java/core/src/main/java/co/worklytics/psoxy/rules/Rules2.java

```diff
@@ -68,9 +69,10 @@ public Rules2 withTransformByEndpoint(String pathRegex, Transform... transforms)
         Endpoint endpoint = matchedEndpoint
             .orElseThrow(() -> new IllegalArgumentException("No endpoint found for pathRegex: " + pathRegex));
 
-        endpoint.transforms = new ArrayList<>(endpoint.getTransforms());
+        endpoint.setTransforms(new ArrayList<>(endpoint.getTransforms()));
 
-        Arrays.stream(transforms).forEach(endpoint.transforms::add);
+        Arrays.stream(transforms)
+            .forEach(endpoint.getTransforms()::add);
```

Comment:
I think in theory should be a `List<>`, because you want a guaranteed order of execution. I haven't thought about it too hard, but in theory I think results will be undefined otherwise - eg `f(g(x)) != g(f(x))` for all transforms `f,g`